golang/server/server.go

```diff
@@ -114,21 +117,45 @@ func (server MinimalGRPCServer) RunUntilStopped(stopper <-chan struct{}) error {
 	}
 
 	grpcServerResultChan := make(chan error)
+	mux := cmux.New(listener)
+	grpcWebL := mux.Match(cmux.HTTP1Fast())
+	grpcL := mux.Match(cmux.Any())
+
+	go func() {
+		if resultErr := grpcServer.Serve(grpcL); resultErr != nil {
+			logrus.Debugf("error ocurred while creating grpc server: %v", resultErr)
+			grpcServerResultChan <- resultErr
+		}
+	}()
+
+	// grpcweb is a proxy which enables ui to make requests to grpc server using kurtosis-sdk
+	// it converts http/1 request to http/2 and vice-versa under the hood
```

Comment:
Is the browser sending http1 or 2?

Reply:
I was just looking at the docs - there is a limitation here where ` Bi-di and client streams are unsupported due to limitations in browser protocol support. https://github.com/improbable-eng/grpc-web/tree/master/go/grpcweb

Is this still the right thing to be using? We have a few client side streaming endpoints like - 

  // Uploads a Starlark package. This step is required before the package can be executed with RunStarlarkPackage
  rpc UploadStarlarkPackage(stream StreamedDataChunk) returns (google.protobuf.Empty) {};


  // Uploads a files artifact to the Kurtosis File System
  rpc UploadFilesArtifact(stream StreamedDataChunk) returns (UploadFilesArtifactResponse) {};


is the proxy a better choice?

Reply:
good point; in the v0 and even v1... don't need bi-directional and client side streaming; I would imagine i probably would incorporate https://connect.build/docs/protocol/ instead of proxy as it support all sorts of streaming.. but for near term don't think urgent need.